internal/mining/mining.go

```diff
@@ -1426,11 +1627,26 @@ nextPriorityQueueItem:
 			continue
 		}
 
-		// Skip all missed tickets that we've never heard of.
+		// Skip revocation transactions if they are spending tickets that are not
+		// eligible to be revoked.
 		if isSSRtx {
 			ticketHash := &tx.MsgTx().TxIn[0].PreviousOutPoint.Hash
 
-			if !hashInSlice(*ticketHash, best.MissedTickets) {
+			// Determine if the ticket being revoked will become missed or expired as
+			// of this block.
+			voted, winning := foundWinningTickets[*ticketHash]
+			missedThisBlock := !voted && winning
+			_, expiringThisBlock := expiringTicketHashes[*ticketHash]
+			missedOrExpiredThisBlock := missedThisBlock || expiringThisBlock
```

Comment:
This loop this is in is kinda huge, so not sure, but this will also go through tickets pushed to the queue in the other added if block?

Reply:
Correct, this logic is hit for standard revocations as well as any automatic revocations that are added.